gensim/similarity_learning/evaluation_scripts/d2v_train_eval.py

```diff
@@ -0,0 +1,128 @@
+from gensim.similarity_learning import WikiQAExtractor
+from gensim.models.doc2vec import TaggedDocument, Doc2Vec
+import os, pickle
+import numpy as np
+import logging
+
+logging.basicConfig(format='%(asctime)s : %(levelname)s : %(message)s', level=logging.INFO)
+
+class LabeledLineSentence(object):
+
+    def __init__(self, corpus):
+        self.corpus = corpus
+
+    def __iter__(self):
+        for uid, line in enumerate(self.corpus):
+            yield TaggedDocument(words=line.split(), tags=['SENT_%s' % uid])
+
+def my_map(big_y_true, big_y_pred):
+    
+    aps = []
+
+    for y_true, y_pred in zip(big_y_true, big_y_pred):
+
+        pred_sorted = sorted(zip(y_true, y_pred), key=lambda x:x[1], reverse=True)
+
+        avg = 0
+        n_relevant = 0
+
+        for i, val in enumerate(pred_sorted):
+            if val[0] == 1:
+                avg += 1./(i + 1.)
+                n_relevant += 1
+
+        if n_relevant != 0:
+            ap = avg/n_relevant
+            aps.append(ap)
+            
+    return np.mean(np.array(aps))
+
+
+def mean_ndcg(big_y_true, big_y_pred, k=10):
+    
+    ndcgs = []
+
+    for y_true, y_pred in zip(big_y_true, big_y_pred):
+
+        pred_sorted = sorted(zip(y_true, y_pred), key=lambda x:x[1], reverse=True)
+        true_sorted = sorted(zip(y_true, y_pred), key=lambda x:x[0], reverse=True)
+
+        pred_sorted = pred_sorted[:k]
+        true_sorted = true_sorted[:k]
+
+        dcg = 0
+
+        for i, val in enumerate(pred_sorted):
+            if val[0] == 1:
+                dcg += 1./np.log2(i + 2)
+
+
+        idcg = 0
+
+        for i, val in enumerate(true_sorted):
+            if val[0] == 1:
+                idcg += 1./np.log2(i + 2)
+
+        if idcg != 0:
+            ndcgs.append(dcg/idcg)
+
+    return np.mean(np.array(ndcgs))
+
+def cos_sim(vec1, vec2):
+    return np.sum(vec1*vec2)/(np.linalg.norm(vec1)*np.linalg.norm(vec2))
+
+wikiqa_train = WikiQAExtractor(os.path.join("..", "data", "WikiQACorpus", "WikiQA-train.tsv"))

```

Comment:
The important point - I can't run this script, because, I have no this files and folder, need to fix this (here and everywhere)